### DISCUSSION TITLE
Looking for help regarding a bash script

### DISCUSSION THREAD

**Discussion by Ellaina22** [community/community#21371] [Other Feature Feedback, Questions, & Ideas]
<p>Hi everyone :slight_smile:</p>
<p>I am looking for some informations regarding a bash script.</p>
<p>I am using the commande lspi | grep -i net to get information toward network components.</p>
<p>My script is the following :</p>
<p><img src="https://user-images.githubusercontent.com/53341239/181063925-a3c26e1f-a9ce-49b8-bfa6-df366b9f25f3.png" alt="Capture du 2019-07-26 14-29-17.png" data-base62-sha1="z8IkI4LZO3JedSNsjCr2vYuPxeU" width="381" height="236"></p>
<p>Rm allows me to reset the text file I am creating each time I run the script (or else it will keep adding in the file text without clearing it) [If you have any ways to do it better, I am taking it !]</p>
<p>This script is suppose to allow me to know if there is a wifi card, or not (and it works that way) and will be a part of a bigger one for several informations, to make a paper that will be automatically printed and fullfilled.</p>
<p>My problem here is that I would like to copy paste to “Network Controller” line and informations into a text file (or have it in any way)</p>
<p>I tried multiple things but Ican’t find any way to do so, atleast from the terminal itself</p>
<p>Would anyone have any tips about doing that ?</p>
<p>Thanks in advance</p>

**Comment by Ellaina22** [comment]
<p>Now using the following line</p>
<p>echo $Wifi &gt; txt.txt</p>
<p>to get the information from the command</p>
<p>But I cant get to know how to select specifically the network controller line only, anyone has a clue for it ?</p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/ellaina22">@ellaina22</a>,</p>
<p>You can replace your script with this:</p>
<pre><code>#!/bin/bash

**Wifi=$(lspci | grep -i net | grep 'Network controller');**

if [[$Wifi == *"Network controller"*]];
then
**echo "|Wifi ON| |" \&gt; wifion.md;**
else
**echo "|NO Wifi| |" \&gt; nowifi.md;**
fi
</code></pre>
<p>the line with the declaration of the Wifi retrive only the <strong>‘Network controller’</strong> line and than inside the if controll statement you need to replace the <strong>&gt;&gt; with only one &gt;</strong>, this rewrite completely the file and you can remove the first two line of remove.</p>
<p>Hope this help, ask if you need farther infos…</p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>
